File: TortoiseHare/TortoiseHare.java
import java.util.*;
interface Func{
	int compute(int x);
}
public class TortoiseHare{
	public static int[] TortoiseHare(Func f,int xo){
		int tortoise = f.compute(xo);
		int hare = f.compute(f.compute(xo));
		while(tortoise!=hare){
			tortoise = f.compute(tortoise);
			hare = f.compute(f.compute(hare));
		}

		//Detect Cycle period
		int[] ans = new int[2];
		ans[0] =0;
		ans[1] =1;
		tortoise = xo;
		while(tortoise!=hare){
			tortoise = f.compute(tortoise);
			hare = f.compute(hare);
			ans[0]++;
		}

		hare = f.compute(tortoise);
		while(tortoise!=hare){
			hare = f.compute(hare);
			ans[1]++;
		}
		return ans;
	}
	public static void main(String[] args){
		int[] ans = TortoiseHare(new Func(){
			public int compute(int x){
				return (3*x*x + 7*x + 5) % 97;
			}
		},62);
		System.out.println(ans[0]+" "+ans[1]);
	}
}

File: TortoiseHare/TortoiseHare.cpp
#include <iostream>
using namespace std;
int f(int x){
	return (3*x*x + 7*x + 5) % 97;
}
pair<int,int> FloydTortoise(int (*f)(int),int xo){
	int tortoise = f(xo);
	int hare = f(f(xo));
	while(tortoise!=hare){
		tortoise = f(tortoise);
        hare = f(f(hare));
	}

	pair<int,int> ans = make_pair(0,1);
	tortoise = xo;
	while(tortoise!=hare){
		tortoise = f(tortoise);
		hare = f(hare);
		ans.first++;
	}

    hare = f(tortoise);
    while(tortoise!=hare){
    	hare = f(hare);
    	ans.second++;
    }
    return ans;
}
int main(){
	pair<int,int> ans = FloydTortoise(f,62);
	cout<<ans.first<<" "<<ans.second<<endl;
}